runner/src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -305,6 +306,68 @@ export class PageControllerBase {
     return request.yar.get("lang");
   }
 
+  calculateRouteState(page, completeState, state = {}) {
```

Comment:
will relying on this.routeState affect other users? This will work on the assumption that it's 1 page instance for one 1 user. What if there are multiple requests coming in at once for /a-page, and this.routeState keeps getting changed

Reply:
Sorted, we're now creating a variable to hold the new state which should sort the issue.